sqlite3.go

```diff
@@ -1977,3 +2006,70 @@ func (rc *SQLiteRows) Next(dest []driver.Value) error {
 	}
 	return nil
 }
+
+// Depth returns the source path of the write, see sqlite3_preupdate_depth()
+func (d *SQLitePreUpdateData) Depth() int {
+	return int(C.sqlite3_preupdate_depth(d.Conn.db))
+}
+
+// Count returns the number of columns in the row
+func (d *SQLitePreUpdateData) Count() int {
+	return int(C.sqlite3_preupdate_count(d.Conn.db))
+}
+
+func (d *SQLitePreUpdateData) row(dest []interface{}, new bool) error {
```

Comment:
Please move this to a file named sqlite3_opt_preupdate_hook.go and set a build tag sqlite_preupdate_hook